# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1374813536 14400
#      Fri Jul 26 00:38:56 2013 -0400
# Node ID 3cbbdb49b8fffe42d6ed19c843ccbbd670e88788
# Parent  c8586e6fc60d05aeff76a9b012af19949f031824
interpft.m: Fix decimation by a non-integer factor (bug #39017)

* interpft.m: Fix decimation when the desired number of points does not
evenly divide the input size.

diff --git a/scripts/general/interpft.m b/scripts/general/interpft.m
--- a/scripts/general/interpft.m
+++ b/scripts/general/interpft.m
@@ -63,17 +63,17 @@ function z = interpft (x, n, dim)
   if (dim < 1 || dim > nd)
     error ("interpft: invalid dimension DIM");
   endif
 
   perm = [dim:nd, 1:(dim-1)];
   x = permute (x, perm);
   m = rows (x);
 
-  inc = max (1, fix (m/n));
+  inc = ceil (m/n);
   y = fft (x) / m;
   k = floor (m / 2);
   sz = size (x);
   sz(1) = n * inc - m;
 
   idx = repmat ({':'}, nd, 1);
   idx{1} = 1:k;
   z = cat (1, y(idx{:}), zeros (sz));
